src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -431,5 +431,11 @@ public override BoundExpression InstrumentSwitchStatementExpression(BoundStateme
             return AddConditionSequencePoint(base.InstrumentSwitchStatementExpression(original, rewrittenExpression, factory), original.Syntax, factory);
         }
 
+        public override BoundStatement InstrumentPatternSwitchBindCasePatternVariables(BoundStatement bindings)
+        {
+            // Mark the code that binds pattern variables to their values as hidden.
+            // We do it to tell that this is not a part of previous statement.
+            return new BoundSequencePoint(null, base.InstrumentPatternSwitchBindCasePatternVariables(bindings));
```

Comment:
Synthetic code reachable via a label has this habit of being attributed to the perevious statement unless shielded from that via a hidden SP.